arch/arm/core/Kconfig

```diff
@@ -95,6 +95,12 @@ config RUNTIME_NMI
 	  NMI handler installed when the CPU boots. If a custom handler is
 	  needed, enable this option and attach it via _NmiHandlerSet().
 
+config PLATFORM_SPECIFIC_INIT
+	bool "Enable platform (SOC) specific startup hook"
+	help
+	  The platform specific initialization code (z_platform_init) is executed
+	  at the beginning of the startup code (__start).
```

Comment:
This also means @stephanosio that you plan to have/keep a  __start symbol in Cortex-R

Reply:
@ioannisg I left it as is for now, because both Cortex-M and Cortex-R still has this symbol (maybe it can be removed/changed when we go on another renaming spree?).